timetracker/entries/models.py

```diff
@@ -2,24 +2,39 @@
 
 from django.utils import timezone
 
+# One Project may have multiple clients, or two Projects may have one client, causing redundancy. To denormalise the table we isolate Project.client from the Projects model by creating a new model of Clients with which projects share a many-many relationship.
+
+class Client(models.Model):
+    name = models.CharField(max_length=200)
+
+    #how to create a method for displaying foreign key attributes? eg: a client's projects?
+
+    def __str__(self):
+        return '{}'.format(self.name)
 
 class Project(models.Model):
-    client = models.CharField(max_length=200)
     name = models.CharField(max_length=200)
+    client = models.ManyToManyField(Client, related_name="+")
+    
+    def for_clients(self):
+        return "\n".join([str(p.name) for p in self.client.all()])
 
     def __str__(self):
-        return '<{}> {}'.format(self.client, self.name)
+        return '{} {}'.format(self.name, self.client)
 
+# Two entries with the same description, (eg: the same work performed on two different days, or performed simultaneously for two different clients), suggests that entries are better conceptualised as 'tasks'. To distinguish tasks from entries, we need to denormalise (2NF?) the Entry table by creating a new table of tasks with a many-many relationship between them.
 
-# Create your models here.
-class Entry(models.Model):
+class Task(models.Model):
     start = models.DateTimeField(default=timezone.now)
     stop = models.DateTimeField(blank=True, null=True)
-    project = models.ForeignKey('Project')
     description = models.CharField(max_length=200)
-
-    def __str__(self):
-        return '[{} - {}] ({}) {}'.format(self.start, self.stop, self.project, self.description)
-
+    project = models.ManyToManyField(Project, related_name="+", blank=True)
     def is_finished(self):
         return self.stop is not None
+    is_finished.boolean = True
```

Comment:
I forgot how to get icons showing in admin.. this is working. Maybe should be in admin.py?